ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -5064,6 +5070,7 @@ private RowResolver getColForInsertStmtSpec(Map<String, ExprNodeDesc> targetCol2
       }
       colListPos++;
     }
+    SessionStateUtil.addResource(conf, MISSING_COLUMNS, missingColumns);
```

Comment:
not sure it's a best place to populate missing columns. this is a getter with hidden side functionality.